scilpy/image/volume_space_management.py

```diff
@@ -361,3 +364,180 @@ def _is_voxmm_in_bound(self, x, y, z, origin):
+
+        if space == Space.VOX:
+            return self._voxmm_to_value(*self.vox_to_voxmm(x, y, z), origin)
+        elif space == Space.VOXMM:
+            return self._voxmm_to_value(x, y, z, origin)
+        else:
+            raise NotImplementedError("We have not prepared the DataVolume "
+                                      "to work in RASMM space yet.")
+
+    def is_idx_in_bound(self, i, j, k):
+        return super().is_idx_in_bound(i, j, k)
+
+    def is_coordinate_in_bound(self, x, y, z, space, origin):
+        self._validate_origin(origin)
+        return super().is_coordinate_in_bound(x, y, z, space, origin)
+
+    # TODO: Figure out how to validate origin within vox_to_idx
+
+    def voxmm_to_idx(self, x, y, z, origin):
+        self._validate_origin(origin)
+        return super().voxmm_to_idx(x, y, z, origin)
+
+    def vox_to_voxmm(self, x, y, z):
+        """
+        Get mm space coordinates at position x, y, z (vox).
+
+        Parameters
+        ----------
+        x, y, z: floats
+            Position coordinate (vox) along x, y, z axis.
+
+        Return
+        ------
+        x, y, z: floats
+            mm space coordinates for position x, y, z.
+        """
+
+        # Does not depend on origin!
+        # In each dimension:
+        #   In corner: 0 to 1 will become 0 to voxres.
+        #   In center: -0.5 to 0.5 will become -0.5*voxres to 0.5*voxres.
+        return [x * self.voxres[0],
+                y * self.voxres[1],
+                z * self.voxres[2]]
+
+    def _clip_voxmm_to_bound(self, x, y, z, origin):
+        return self.vox_to_voxmm(*self._clip_vox_to_bound(
+            *self.voxmm_to_vox(x, y, z), origin))
+
+    def _vox_to_value(self, x, y, z, origin):
+        return self._voxmm_to_value(*self.vox_to_voxmm(x, y, z), origin)
+
+    def _voxmm_to_value(self, x, y, z, origin):
+        x, y, z = self._clip_voxmm_to_bound(x, y, z, origin)
+
+        pos = np.array([x, y, z], dtype=np.float64)
+
+        neighbors = self.tree.query_radius(
+            pos, self.sampling_radius + self.max_seg_length / 2)[0]
+
+        return self.extract_directions(pos, neighbors, self.sampling_radius,
+                                       self.segments_indices, self.fibers,
+                                       self.diameters, self.random_generator)
+
+    def get_absolute_direction(self, x, y, z):
+        pos = np.array([x, y, z], np.float64)
+
+        neighbors = self.tree.query_radius(
+            pos, self.sampling_radius + self.max_seg_length / 2)[0]
+
+        for segi in neighbors:
+            fi, pi = self.segments_indices[segi]
+            fiber = self.fibers[fi]
+            radius = self.diameters[fi] / 2
+
+            if point_in_cylinder(fiber[pi], fiber[pi+1], radius, pos):
+                return fiber[pi+1] - fiber[pi]
+
+        return None
+
+    @staticmethod
+    @njit
+    def extract_directions(pos, neighbors, sampling_radius, segments_indices,
```

Comment:
can you comment the different steps in this method?

Reply:
And explain why static